CMake/resolve_dependency_modules/protobuf.cmake

```diff
@@ -13,15 +13,15 @@
 # limitations under the License.
 include_guard(GLOBAL)
 
-set(VELOX_PROTOBUF_BUILD_VERSION 21.8)
+set_source(absl)
+resolve_dependency(absl 20240116 EXACT)
+
+set(VELOX_PROTOBUF_BUILD_VERSION 27.1)
```

Comment:
Does that mean we will now require a newer version of protobuf - meaning 21.8 will now be unsupported? Do we know which version do most users still use? For example, which protobuf version ships with current popular distros like Ubuntu, Fedora, Centos, etc?

Reply:
Keep in mind that it is more of a concept so it won't be submitted as final.
The first commit adds backwards and forward compatibility, and I think it can be submitted as a different PR.
I added the second commit precisely to test that both protobuf versions are supported, and it should be submitted as another PR as needed, or maybe pick the versions from a manifest, I don't know what is best.

Reply:
In the last PR, now in its final form, we only have to modify the .cmake modules and the setup-centos, setup-adapters files. I would like to centralize the versioning of the dependencies but that may be a future improvement.